doc/source/cli.md

```diff
@@ -0,0 +1,407 @@
+
+            The data provided in extra_information is not validated beforehand.
+            Its use is entirely the responsibility of the user.
+
+      .. code-block:: yaml
+
+        coregistration:
+          step_one:
+            method: "NuthKaab"
+            extra_informations : {"max_iterations": 10}
+          step_two:
+            method: "DHMinimize"
+          sampling_grid: "reference_elev"
+          process: True
+
+   .. tab:: statistics
+
+      **Required:** No
+
+      Statistics step information. This section relates to the computed statistics:
+        1. If no block is specified, all available statistics are calculated by default.
+        2. If a block is specified but no statistics are provided, then no statistics will be computed.
+
+      .. code-block:: yaml
+
+        statistics:
+          - min
+          - max
+          - mean
+
+   .. tab:: outputs
+
+     **Required:** No
+
+     Outputs information.
+     Operates by levels:
+     1. Level 1 → aligned elevation only
+     2. Level 2 → more detailed output
+
+     .. csv-table:: Outputs parameters
+       :header: "Name", "Description", "Type", "Default value", "Available Value", "Required"
+       :widths: 20, 40, 10, 10, 10, 10
+
+       "path", "Path for outputs", "str", "outputs", "", "No"
+       "level", "Level for detailed outputs", "int", "1", "1 or 2", "No"
+       "output_grid", "Grid for outputs resampling", "str", "reference_elev", "reference_elev or to_be_aligned_elev", "No"
+
+     .. code-block:: yaml
+
+       outputs:
+           level : 1
+           path : "path_to/outputs"
+           output_grid: "reference_elev"
+
+     Tree of outputs for level 1 (including coregistration step)
+
+     .. code-block:: text
+
+      - root
+        ├─ csv
+        │   └─ aligned_elev_stats.csv
+        ├─ png
+        │   ├─ diff_elev_after_coreg_map.png
+        │   ├─ diff_elev_before_coreg_map.png
+        │   ├─ diff_elev_before_after_hist.png
+        │   ├─ reference_elev_map.png
+        │   └─ to_be_aligned_elev_map.png
+        ├─ raster
+        │   └─ aligned_elev.tif
+        ├─ report.html
+        ├─ report.pdf
+        └─ used_config.yaml
+
+     Tree of outputs for level 2 (including coregistration step)
+
+     .. code-block:: text
+
+      - root
+        ├─ csv
+        │   ├─ aligned_elev_stats.csv
+        │   ├─ diff_elev_after_coreg_stats.csv
+        │   ├─ diff_elev_before_coreg_stats.csv
+        │   ├─ reference_elev_stats.csv
+        │   └─ to_be_aligned_elev_stats.csv
+        ├─ png
+        │   ├─ diff_elev_after_coreg_map.png
+        │   ├─ diff_elev_before_coreg_map.png
+        │   ├─ diff_elev_before_after_hist.png
+        │   ├─ reference_elev_map.png
+        │   └─ to_be_aligned_elev_map.png
+        ├─ raster
+        │   ├─ aligned_elev.tif
+        │   ├─ diff_elev_after_coreg.tif
+        │   ├─ diff_elev_before_coreg.tif
+        │   └─ to_be_aligned_elev_reprojected.tif
+        ├─ report.html
```

Comment:
To echo @belletva's comment during our last call on the directory names: I think we should either go for extension names for all (`csv`, `png`, `tif`) or for data type names for all (`tables`, `plots`, `rasters`).
My preference would be for the second, I think it's more explicit to users of any level!